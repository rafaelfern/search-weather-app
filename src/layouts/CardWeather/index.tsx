import React, { useEffect, useState } from 'react'
import { IWeatherProps } from '@/pages/utils/types'
import Image  from 'next/image'
import { FaSun } from "react-icons/fa";

export interface ICardWeatherProps {
    weatherData: IWeatherProps | undefined
}

export default function CardWeather({ weatherData }: ICardWeatherProps) {
    
    // const verifyCardColorByIcon = (icon: string) => {
    //     const colorCard:any = {
    //         '02d': '#983628',
    //         '01d': '#D8BD8A',
    //         '04d': '#9ca3af',
    //         '10n': '#374151',
    //         '50n': '#fca5a5',
    //         'default': '#86efac'
    //     }

    //     return `bg-[${colorCard[icon]}]` || `bg-[${colorCard.default}]`
    // }
// console.log("+++++",verifyCardColorByIcon('10n'))
    return (
        <>  
            {weatherData &&
                <div className={`card ${weatherData.weather[0].color} text-white w-[600px] h-[90px] flex flex-row justify-start items-center mt-10 rounded`}>
                    <Image 
                        src={`http://openweathermap.org/img/w/${weatherData.weather[0].icon}.png`}
                        alt=""
                        width={50}
                        height={50}
                    />
                    <div className='flex flex-col ml-4'>
                        <h4 className='text-4xl'>{weatherData.weather[0].description}</h4>
                        <div className='mt-2 flex flex-row'>
                            <h2 className='text-1xl font-bold mb-2'>{weatherData.main.temp} ÂºC</h2>
                            <div className="ml-3 border-l-4 flex items-center w-100">
                                <FaSun className='ml-1 mr-1'/>
                                <h6 >{weatherData.sys.sunrise}</h6> - 
                                <h6 >{weatherData.sys.sunset}</h6>
                            </div>
                        </div>
                    </div>
                </div>
            }
        </>
    )
}
