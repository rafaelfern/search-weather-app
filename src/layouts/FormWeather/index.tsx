import React, { LegacyRef } from 'react'

export interface IFormWeatherProps {
    handleSubmit: (inputValue: any) => void
    inputRef: LegacyRef<HTMLInputElement> | undefined
}

export default function FormWeather({ handleSubmit, inputRef}: IFormWeatherProps) {
  return (
    <>
    <h1 className="title mb-6 mt-4 py-4 text-5xl font-serif">Search Weather</h1>
        <div className="form">
            <form onSubmit={handleSubmit}>
                <input 
                    ref={inputRef}
                    type="text" 
                    placeholder='Enter city name'
                    className="px-4 py-3 rounded-sm mr-2"
                    onChange={() => {}}
                />
                <button type="submit" className='btn px-4 py-3 text-white rounded-sm'>
                    Search
                </button>
            </form>
        </div>
    </>
  )
}
