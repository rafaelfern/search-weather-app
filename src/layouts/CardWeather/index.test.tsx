import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import FormWeather, { ICardWeatherProps } from './index'
import CardWeather from './index'
import fetchMock from 'jest-fetch-mock';
import axios from "axios";

const testWeatherDataResponse: any = {
    coord: { lon: -43.2075, lat: -22.9028 },
    weather: [
        {
        id: 802,
        main: 'Clouds',
        description: 'nuvens dispersas',
        icon: '03d'
        }
    ],
    base: 'stations',
    main: {
        temp: 25.23,
        feels_like: 26.08,
        temp_min: 24.98,
        temp_max: 28.97,
        pressure: 1015,
        humidity: 87
    },
    visibility: 7000,
    wind: { speed: 2.57, deg: 10 },
    clouds: { all: 40 },
    dt: 1679658432,
    sys: {
        type: 1,
        id: 8429,
        country: 'BR',
        sunrise: '07:34',
        sunset: '07:34'
    },
    timezone: 7200,
    id: 3451190,
    name: 'Rio de Janeiro',
    cod: 200
    
}

describe('<CardWeather />', () => {
    jest.mock("axios");
    
    it('should renders weather data when API call succeeds', async () => {
        const data = testWeatherDataResponse
        axios.post = jest.fn().mockResolvedValue({ data })
        const response = await axios.post('/api/weather', 'Rio de Janeiro')
        expect(response.data.name).toEqual(data.name)
    })

    it('should renders error when API call fails', async () => {

    })
})