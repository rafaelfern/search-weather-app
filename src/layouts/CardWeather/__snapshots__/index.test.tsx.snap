// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWeather /> it should render card component with props 1`] = `
<div>
  <div
    class="card bg-purple-500 text-white w-[600px] h-[90px] flex flex-row justify-start items-center mt-10 rounded"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fopenweathermap.org%2Fimg%2Fw%2F03d.png&w=128&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fopenweathermap.org%2Fimg%2Fw%2F03d.png&w=64&q=75 1x, /_next/image?url=http%3A%2F%2Fopenweathermap.org%2Fimg%2Fw%2F03d.png&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
    <div
      class="flex flex-col ml-4"
    >
      <h4
        class="text-4xl"
      >
        nuvens dispersas
      </h4>
      <div
        class="mt-2 flex flex-row"
      >
        <h2
          class="text-1xl font-bold mb-2"
        >
          25.23
           ÂºC
        </h2>
        <div
          class="ml-3 border-l-4 flex items-center w-100"
        >
          <svg
            class="ml-1 mr-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
            />
          </svg>
          <h6>
            07:34
          </h6>
           -
          <h6>
            07:34
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
